lib/components/RangePicker.vue

```diff
@@ -0,0 +1,326 @@
+<script lang="ts">
+import { defineComponent, VNode, DirectiveBinding, PropType } from 'vue'
+import { faGripLinesVertical } from '@fortawesome/free-solid-svg-icons/faGripLinesVertical'
+import { clamp, get, has, invoke, round } from 'lodash'
+
+import Fa, { library } from './Fa'
+
+import type { Variant } from '@/types'
+
+/**
+ * A component to wrap an HTML element with a range picker overlay.
+ */
+export default defineComponent({
+  name: 'RangePicker',
+  components: {
+    Fa
+  },
+  directives: {
+    draggable: {
```

Comment:
I'm not familiar with directives: are the event listeners applied on the $el (like mousedown) removed on destroy?

Reply:
Yep, event listeners are garbage collected by the web browser when destroying the element. So no need to destroy them explicitly.

Reply:
Neat! thanks